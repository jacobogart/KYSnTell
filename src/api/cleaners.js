export const locationCleaner = (location) => {
  const { title, link, id, streetAddress, locality, region, postalCode, telephone, point} = location;
  const shortId = id.split('=')[1];
  const address = `${streetAddress};${locality}, ${region} ${postalCode}`;
  return {
    title,
    link,
    id: shortId,
    address,
    telephone,
    point
  }
}