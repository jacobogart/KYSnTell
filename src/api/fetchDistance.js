import { googleKey, googleRoot } from "./utilities";

export const fetchDistance = (origin, destination) => {  return fetch(`${googleRoot}distancematrix/json?origins=${origin.lat},${origin.lng}&destinations=${destination.lat},${destination.long}&units=imperial&key=${googleKey}`)
    .then(response => {
      if(!response.ok) {
        throw Error('Failed to find distance')
      } else {
        return response.json()
      }
    })
    .then(result => result.rows[0].elements[0].distance.text)
}