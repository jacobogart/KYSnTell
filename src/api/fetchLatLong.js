import { geocodeUrl } from "./utilities";

export const fetchLatLong = (zip) => {
  const googleKey = process.env.REACT_APP_GOOGLE_KEY;
  return fetch(`${geocodeUrl}${zip}&key=${googleKey}`)
    .then(response => {
      if(!response.ok) {
        throw Error('Failed to find coordinates')
      } else {
        return response.json()
      }
    })
    .then(result => result.results[0].geometry.location)
}