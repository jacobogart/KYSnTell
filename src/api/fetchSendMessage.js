export const fetchSendMessage = (contacts, message) => {
  const messagePromises = contacts.map(contact => {
      const body = { to: contact, message}
      const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(body)
    }
    return fetch('/api/messages', options)
      .then(response => {
        if (!response.ok) {
          throw Error('Failed to send message')
        } else {
          return response.json();
        }
      })
    })
  return Promise.all(messagePromises)
}
  