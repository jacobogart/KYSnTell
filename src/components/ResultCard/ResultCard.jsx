import React from 'react';
import { Link } from 'react-router-dom'

export const ResultCard = (props) => {
  const { id, title, address } = props
  return (
    <Link to={`/kys/locations/${id}`}>
      <article>
        <h3>{title}</h3>
        <p>{address}</p>
      </article>
    </Link>
  )
}
