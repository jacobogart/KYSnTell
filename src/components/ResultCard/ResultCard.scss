.ResultCard {
  background-color: white;
  margin: 15px;
  border-radius: 10px;
  padding: 5px;
  text-decoration: none;
  height: 100%;
  width: 90vw;
  article {
    display: flex;
    flex-direction: row;
    position: relative;
  }
  
  
  h3,
  p {
    margin: 10px;
  }
}

.card-info {
  width: 82%;
  color: black;
}

.distance-container {
  padding: 5px;
  text-align: center;
  background: grey;
  height: 100%;
  width: 46px;
  border-radius: 0 10px 10px 0;
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  right: -5px;
  top: -5px;

  p {
    height: 45px;
  }
}