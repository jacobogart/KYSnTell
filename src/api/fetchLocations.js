import { url } from "./utilities";
import { locationCleaner } from "./cleaners";

export const fetchLocations = (location, distance) => {
  const { lat, lng } = location
  const fullUrl = `${url}lat=${lat}&long=${lng}`
  const distanceURL = distance 
    ? `&distance=${distance}` 
    : '';
  return fetch(`${fullUrl}${distanceURL}`)
    .then(response => {
      if(!response.ok) {
        throw Error('Failed to find locations')
      } else {
        return response.json();
      }
    })
    .then(results => results.services[0].providers.map(location => locationCleaner(location)))
    .catch(error => console.log(error))
}