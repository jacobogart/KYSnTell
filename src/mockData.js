import { locationCleaner } from "./api/cleaners";

export const origin = {
  lat: 38.9119463, 
  lng: -94.4136213 
}

export const destination = { 
  lat: "39.1952922", 
  lng: "-94.6817269" 
}

export const distance = '18.3 mi';

export const distanceResult = {
  rows: [
    {
      elements: [
        {
          distance: {
            text: distance
          }
        }
      ]
    }
  ]
}


export const fetchDistanceURL = "https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/distancematrix/json?origins=38.9119463,-94.4136213&destinations=39.1952922,-94.6817269&units=imperial&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const latLongResult = {
  results: [
    {
      geometry: {
        location: origin
      }
    }
  ]
}


export const fetchLatLongURL = "https://maps.googleapis.com/maps/api/geocode/json?address=64081&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const locations = [
  {
    title: "Neighborhood Health",
    link: "https://gettested.cdc.gov/gettested_redirect/21952",
    id: "tag:hivtest.org,2000-03-20:/search/Detailx.asp?id=21952",
    streetAddress: "617 S 8th St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206-3819",
    telephone: "(615)-227-3000",
    categories: "",
    point: {
      lat: "36.1678930",
      long: "-86.7547047"
    },
    updated: "2019-02-08T17:19:32.0000000-05:00"
  },
  {
    title: "Street Works",
    link: "https://gettested.cdc.gov/gettested_redirect/110792",
    id: "tag:hivtest.org,2008-05-21:/search/Detailx.asp?id=110792",
    streetAddress: "520 Sylvan St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206",
    telephone: "(615)-259-7676",
    categories: "",
    point: {
      lat: "36.1665114",
      long: "-86.7605114"
    },
    updated: "2018-08-02T18:05:30.0000000-04:00"
  }
];

export const locationsResult = {
  services: [
    {
      providers: locations
    }
  ]
}

export const cleanLocations = locations.map(location => locationCleaner(location))

export const fetchLocationsURL = "https://cors-anywhere.herokuapp.com/https://locator.hiv.gov/data?lat=38.9119463&long=-94.4136213&distance=10"

export const contact = '+18168682142';

export const contacts = [contact, contact]

export const message = 'Test message.'

export const sendMessageOptions = {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({ to: contact, message })
}

export const resolvedMessages = [{ ok: true }, { ok: true }];

export const cleanLocation = {
  title: 'Neighborhood Health',
  link: 'https://gettested.cdc.gov/gettested_redirect/21952',
  id: '21952',
  address: '617 S 8th St;Nashville, TN 37206-3819',
  telephone: '(615)-227-3000',
  point: { lat: '36.1678930', lng: '-86.7547047' }
}

export const details = {
  additionalNotes: "Test",
  diagnosis: "Herpes",
  timeFrame: "year"
}

export const user = {
  location: origin
}

export const searchChangeEvent = {
  target: {
    name: 'zipcode',
    value: '64081'
  }
}

export const submitEvent = {
  preventDefault: jest.fn()
}

export const history = {
  push: jest.fn()
}

export const setUserLocation = jest.fn();

export const setLocations = jest.fn();

export const setContacts = jest.fn();

export const fullMessage = 'Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider. Additional notes from partner: Test'

export const messageWONotes = "Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider."

export const previewEvent = {
  target: {
    blur: jest.fn()
  }
}

export const messageContext = {
  focus: jest.fn(),
  blur: jest.fn(),
  innerText: 'Test'
}

 export const rejectedMessages = [{ ok: false, contact }]

 export const routerLocation = {
  pathname:'/kys/locations/21952'
 }

export const detailsChangeEvent = {
  target: {
    name: 'diagnosis',
    value: 'Herpes'
  }
}

export const setDetails = jest.fn();

export const contactChangeEvent = {
  target: {
    value: contact
  }
}

export const contactBadChangeEvent = {
  target: {
    value: 'test'
  }
}

export const searchDefaultState = {
  zipcode: '',
  distance: '',
  isLoading: false,
  error: false
}