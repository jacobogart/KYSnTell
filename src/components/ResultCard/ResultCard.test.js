import React from 'react';
import { shallow } from 'enzyme';
import { ResultCard } from './ResultCard';
import * as mock from '../../mockData';
import { fetchDistance } from '../../api/fetchDistance';

jest.mock('../../api/fetchDistance');

describe('ResultCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ResultCard
        {...mock.cleanLocation}
        user={mock.user}
      />);
    });

    fetchDistance.mockImplementation(() => Promise.resolve(mock.distance))
  
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const defaultState = { distance: '' };

    wrapper = shallow(
      <ResultCard
        {...mock.cleanLocation}
        user={mock.user}
      />, { disableLifecycleMethods: true });
    expect(wrapper.state()).toEqual(defaultState);
  });

  it('should invoke fetchDistance on mount', () => {
    const args = [mock.user.location, mock.cleanLocation.point]
    expect(fetchDistance).toHaveBeenCalledWith(...args);
  });

  it('should set state to the correct distance on mount', () => {
    expect(wrapper.state('distance')).toEqual(mock.distance)
  });
});