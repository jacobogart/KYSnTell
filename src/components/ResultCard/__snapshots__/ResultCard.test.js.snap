// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard should match snapshot 1`] = `
<Link
  className="ResultCard"
  to="/kys/locations/21952"
>
  <article>
    <div
      className="card-info"
    >
      <h3>
        Neighborhood Health
      </h3>
      <p>
        617 S 8th St
         
        <br />
         
        Nashville, TN 37206-3819
      </p>
    </div>
    <div
      className="distance-container"
    >
      <p>
        18.3 mi
      </p>
    </div>
  </article>
</Link>
`;
